### ISSUE TITLE
Logging SQL queries and results

### ISSUE THREAD

**Issue by MeanSquaredError** [taocpp/taopq#78]
Hi,

I have a project that uses taopq and I am looking for a way to add logging of the PostgreSQL queries/results that taopq emits.

I checked the code but did not see a way to log the generated SQL queries and/or the returned results, e.g. through a calling callback. In case I am not missing something, are there any plans to add logging to the library in the future?

**Comment by d-frey** [comment]
You are correct, currently there are no logging callbacks available. I hope to add some in the future, but I haven't decided on a specific interface and the exact points when a callback is called yet. Any input is appreciated, feel free to list your ideas and what callbacks you envision.

**Comment by anders-wind** [comment]
Even after overwriting all the log functions I still with the examples from test/log I still get a print:
`NOTICE:  relation "my_table" already exists, skipping`
Perhaps there are still a few places which need to use the logger?

**Comment by d-frey** [comment]
This looks like an exception, so maybe you didn't catch it and by default it's what()-message is printed? Hard to tell without a minimal but complete example.

**Comment by anders-wind** [comment]
I'm quite sure it's not an exception but I'll try to make a complete example tomorrow 👍

**Comment by d-frey** [comment]
Any update on this one?

**Comment by anders-wind** [comment]
I don't currently have the time to create a reproducible build - but I can say it happened inside a `tao::pq::connection::execute` call with a "CREATE TABLE IF NOT EXISTS table_name". 
It was not an exception (I could control/catch) because the rest of my code afterwards continued just fine. I've just worked around it by first checking whether the table exists.
